scripts/build_plugin.py

```diff
@@ -0,0 +1,484 @@
+                    env.get_app_descriptor_dict(eng, app),
+                    fallback_roots=[bundle_cache_root]
+                )
+                logger.info("Caching %s..." % desc)
+                desc.clone_cache(bundle_cache_root)
+
+        for framework in env.get_frameworks():
+            desc = create_descriptor(
+                sg_connection,
+                Descriptor.FRAMEWORK,
+                env.get_framework_descriptor_dict(framework),
+                fallback_roots=[bundle_cache_root]
+            )
+            logger.info("Caching %s..." % desc)
+            desc.clone_cache(bundle_cache_root)
+
+def _process_configuration(sg_connection, source_path, target_path, bundle_cache_root, manifest_data):
+    """
+    Given data in the plugin manifest, download resolve and
+    cache the configuration.
+
+    :param sg_connection: Shotgun connection
+    :param manifest_data: Manifest data as a dictionary
+    :param source_path: Root path of plugin source.
+    :param target_path: Build target path
+    :param bundle_cache_root: Bundle cache root
+    :return: Resolved config descriptor object
+    """
+    logger.info("Analyzing configuration")
+
+    base_config_def = manifest_data["base_configuration"]
+
+    if isinstance(base_config_def, str):
+        # convert to dict so we can introspect
+        base_config_def = descriptor_uri_to_dict(base_config_def)
+
+    # special case - check for the 'baked' descriptor type
+    # and process it
+    if base_config_def["type"] == "baked":
+        logger.info("Baked descriptor detected.")
+
+        baked_path = os.path.expanduser(os.path.expandvars(base_config_def["path"]))
+
+        # if it's a relative path, expand it
+        if not os.path.isabs(baked_path):
+            full_baked_path = os.path.abspath(os.path.join(source_path, baked_path))
+
+            # if it's a relative path, we have already copied it to the build
+            # target location. In this case, attempt to locate it and remove it.
+            baked_target_path = os.path.abspath(os.path.join(target_path, baked_path))
+            if baked_target_path.startswith(baked_target_path):
+                logger.debug("Removing '%s' from build" % baked_target_path)
+                shutil.rmtree(baked_target_path)
+        else:
+            # path is absolute
+            full_baked_path = os.path.abspath(baked_path)
+
+        logger.info("Will bake config from '%s'" % full_baked_path)
+
+        manual_location = os.path.join(bundle_cache_root, "manual", BAKED_BUNDLE_NAME, BAKED_BUNDLE_VERSION)
+        logger.info("Copying %s -> %s" % (full_baked_path, manual_location))
+        filesystem.ensure_folder_exists(manual_location)
+        filesystem.copy_folder(full_baked_path, manual_location)
+
+        # now make a manual descriptor
+        descriptor = {
+            "type": "manual",
+            "version": BAKED_BUNDLE_VERSION,
+            "name": BAKED_BUNDLE_NAME
```

Comment:
the baked bundle name is just an internal thing. It's quite possible that it will change when i look at that spike which looks at "pre-building" the cached configs so that you don't need to copy content across into the tk cache location all the time. the `BAKED_BUNDLE_NAME` is local only to the plugin and is kind of a private thing.

Example from flame taap prototype:
![image](https://cloud.githubusercontent.com/assets/337710/17537348/cd6e8550-5e93-11e6-9865-89f6a8a920a8.png)